nodl_python/nodl/_parsing/_v1/_parsing.py

```diff
@@ -95,11 +111,12 @@ def _parse_node(node: etree._Element) -> Node:
     for child in node:
         if child.tag == 'action':
             actions.append(_parse_action(child))
-        if child.tag == 'parameter':
+        elif child.tag == 'parameter':
             parameters.append(_parse_parameter(child))
-        if child.tag == 'service':
+        elif child.tag == 'service':
             services.append(_parse_service(child))
-        if child.tag == 'topic':
+        else:
+            assert child.tag == 'topic'
```

Comment:
Asserts are opaque, unfriendly, and shouldn't be in production code. Let's have proper error handling instead: please create a new exception for this situation with a descriptive message and raise it.

Reply:
This should actually never fail. The schema prevents this from occurring. The assertion is there just in case something has gone seriously wrong

Reply:
> The assertion is there just in case something has gone seriously wrong

If it does, you end up with a super confusing error. Having an exception that is hopefully never raised is better than an assert. I'm all for belts and suspenders, but there's no reason they can't be user-friendly.

Reply:
I've added it as an `nodl.errors.InvalidElementError` (the base class for parsing errors), should it actually raise a `RuntimeError() from InvalidElementError(...)`?

Reply:
a558e9a

Reply:
Putting it in `nodl.errors` is perfect, although I think it needs to be its own exception, not a generic one with a custom message.